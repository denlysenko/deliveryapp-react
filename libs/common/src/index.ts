export * from './lib/constants';
export * from './lib/errors';
export * from './lib/roles';
export * from './lib/app-menu';
export * from './lib/messages';
export * from './lib/payment-methods';
