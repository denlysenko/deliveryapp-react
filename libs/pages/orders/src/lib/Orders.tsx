import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';

import { Button } from 'primereact/button';
import { Column } from 'primereact/column';
import { ColumnGroup } from 'primereact/columngroup';
import { DataTable } from 'primereact/datatable';
import { PageState, Paginator } from 'primereact/paginator';
import { Row } from 'primereact/row';

import dayjs from 'dayjs';
import { isNil } from 'lodash-es';

import { ORDER_STATUSES, Roles } from '@deliveryapp/common';
import {
  Order,
  OrdersActionTypes,
  ordersClient,
  OrdersFilter as IOrdersFilter,
  useAuth,
  useOrders,
  User
} from '@deliveryapp/data-access';
import { FullPageSpinner } from '@deliveryapp/ui';
import { getSortField, getSortOrder, toCurrency } from '@deliveryapp/utils';

import { OrdersFilter } from './OrdersFilter/OrdersFilter';
import { StyledOrders } from './StyledOrders';

const headerGroup = (user: User | null) => (
  <ColumnGroup>
    <Row>
      <Column header="Num." field="id" rowSpan={2} sortable className="id" />
      <Column
        header="Cargo Name"
        field="cargoName"
        rowSpan={2}
        sortable
        className="cargoName"
      />
      <Column header="Weight/Volume" rowSpan={2} />
      <Column header="Route" colSpan={2} />
      <Column header="Delivery Costs" rowSpan={2} />
      <Column header="Delivery Date" rowSpan={2} />
      <Column header="Status" rowSpan={2} />
      <Column header="Paid" rowSpan={2} />
      {user?.role !== Roles.CLIENT && <Column header="Client" rowSpan={2} />}
    </Row>
    <Row>
      <Column header="From" field="cityFrom" sortable className="cityFrom" />
      <Column header="To" field="cityTo" sortable className="cityTo" />
    </Row>
  </ColumnGroup>
);

const idTemplate = (rowData: Order) => (
  <Link to={`/orders/${rowData.id}`}>{rowData.id}</Link>
);

const weightTemplate = (rowData: Order) => (
  <span>{`${rowData.cargoWeight} kg / ${rowData.cargoVolume || '-'} m3`}</span>
);

const statusTemplate = (rowData: Order) => (
  <span>{ORDER_STATUSES[rowData.status]}</span>
);

const paidTemplate = (rowData: Order) => (
  <span>{rowData.paid ? 'Yes' : 'No'}</span>
);

const deliveryCostsTemplate = (rowData: Order) => (
  <span>
    {isNil(rowData.deliveryCosts) ? '' : toCurrency(rowData.deliveryCosts)}
  </span>
);

const deliveryDateTemplate = (rowData: Order) => (
  <span>
    {isNil(rowData.deliveryDate)
      ? ''
      : dayjs(rowData.deliveryDate).format('DD.MM.YYYY')}
  </span>
);

const clientTemplate = (rowData: Order) => (
  <a href="/">{rowData.client?.email}</a>
);

export const Orders = () => {
  const [{ user }] = useAuth();
  const [ordersFilter, dispatch] = useOrders();
  const [loading, setLoading] = useState(false);
  const [orders, setOrders] = useState<Order[]>([]);
  const [totalRecords, setTotalRecords] = useState(0);

  const doFiltering = (ordersFilter: IOrdersFilter['filter']) => {
    dispatch({
      type: OrdersActionTypes.FILTER_CHANGE,
      payload: ordersFilter
    });
  };

  const doSorting = ({
    sortField,
    sortOrder
  }: {
    sortField: string;
    sortOrder: number;
  }) => {
    dispatch({
      type: OrdersActionTypes.SORTING_CHANGE,
      payload: { [sortField]: sortOrder === 1 ? 'asc' : 'desc' }
    });
  };

  const doPaging = ({ rows, page }: PageState) => {
    dispatch({
      type: OrdersActionTypes.PAGE_CHANGE,
      payload: { limit: rows, offset: rows * page }
    });
  };

  useEffect(() => {
    setLoading(true);

    ordersClient.getOrders(ordersFilter).then(({ data: { rows, count } }) => {
      setOrders(rows);
      setTotalRecords(count);
      setLoading(false);
    });
  }, [ordersFilter]);

  return (
    <>
      {loading ? (
        <FullPageSpinner />
      ) : (
        <StyledOrders>
          <div className="card">
            <div className="p-grid">
              <div className="p-md-8 p-col-12">
                <OrdersFilter
                  filter={ordersFilter.filter}
                  handleFilterChange={doFiltering}
                />
              </div>
              <div className="button p-md-4 p-col-12">
                <Link to="/orders/create">
                  <Button
                    type="button"
                    label="Create order"
                    icon="fa fa-plus"
                    iconPos="left"
                    className="raised-btn"
                  />
                </Link>
              </div>
            </div>
            <DataTable
              value={orders}
              headerColumnGroup={headerGroup(user)}
              sortField={getSortField(ordersFilter.order)}
              sortOrder={getSortOrder(ordersFilter.order)}
              onSort={doSorting}
            >
              <Column body={idTemplate} />
              <Column field="cargoName" />
              <Column body={weightTemplate} />
              <Column field="cityFrom" />
              <Column field="cityTo" />
              <Column body={deliveryCostsTemplate} />
              <Column body={deliveryDateTemplate} />
              <Column body={statusTemplate} />
              <Column body={paidTemplate} />
              {user?.role !== Roles.CLIENT && <Column body={clientTemplate} />}
            </DataTable>
            <Paginator
              rows={ordersFilter.limit}
              totalRecords={totalRecords}
              first={ordersFilter.offset}
              pageLinkSize={3}
              onPageChange={doPaging}
            />
          </div>
        </StyledOrders>
      )}
    </>
  );
};
