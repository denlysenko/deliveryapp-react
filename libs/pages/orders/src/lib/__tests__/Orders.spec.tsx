import React from 'react';
import { fireEvent, screen, waitFor } from '@testing-library/react';

import { DEFAULT_LIMIT } from '@deliveryapp/common';
import {
  Order,
  ordersClient,
  OrdersProvider,
  AuthProvider
} from '@deliveryapp/data-access';
import { renderWithRouter, savedOrder, user } from '@deliveryapp/testing';

import { Orders } from '../Orders';

jest.mock('@deliveryapp/data-access', () => ({
  ...jest.requireActual('@deliveryapp/data-access'),
  useAuth: jest.fn().mockImplementation(() => [{ user }, jest.fn()])
}));

const orders: Order[] = [];

for (let i = 0; i < 12; i++) {
  orders.push(savedOrder);
}

describe('Orders Page', () => {
  beforeEach(() => {
    jest.spyOn(ordersClient, 'getOrders').mockResolvedValue({
      data: { rows: orders, count: 11 }
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render successfully', async () => {
    const { baseElement } = renderWithRouter(
      <AuthProvider>
        <OrdersProvider>
          <Orders />
        </OrdersProvider>
      </AuthProvider>
    );
    await waitFor(() => expect(ordersClient.getOrders).toBeCalledTimes(1));
    expect(baseElement).toMatchSnapshot();
  });

  it('should get orders with default filter', async () => {
    renderWithRouter(
      <AuthProvider>
        <OrdersProvider>
          <Orders />
        </OrdersProvider>
      </AuthProvider>
    );

    await waitFor(() => expect(ordersClient.getOrders).toBeCalledTimes(1));

    expect(ordersClient.getOrders).toBeCalledWith({
      filter: {},
      limit: DEFAULT_LIMIT,
      offset: 0,
      order: {
        id: 'desc'
      }
    });
  });

  describe('filtering', () => {
    it('should filter orders', async () => {
      renderWithRouter(
        <AuthProvider>
          <OrdersProvider>
            <Orders />
          </OrdersProvider>
        </AuthProvider>
      );
      await waitFor(() => expect(ordersClient.getOrders).toBeCalledTimes(1));

      jest.clearAllMocks();

      fireEvent.change(screen.getByPlaceholderText(/search/i), {
        target: {
          value: 'test'
        }
      });

      await waitFor(() => {
        expect(ordersClient.getOrders).toBeCalledWith({
          filter: {
            id: 'test'
          },
          limit: DEFAULT_LIMIT,
          offset: 0,
          order: {
            id: 'desc'
          }
        });
      });
    });
  });

  describe('sorting', () => {
    it('should filter orders by id', async () => {
      renderWithRouter(
        <AuthProvider>
          <OrdersProvider>
            <Orders />
          </OrdersProvider>
        </AuthProvider>
      );
      await waitFor(() => expect(ordersClient.getOrders).toBeCalledTimes(1));

      jest.clearAllMocks();

      fireEvent.click(screen.getByText('Num.'));

      await waitFor(() => {
        expect(ordersClient.getOrders).toBeCalledWith({
          filter: {},
          limit: DEFAULT_LIMIT,
          offset: 0,
          order: {
            id: 'asc'
          }
        });
      });
    });

    it('should filter orders by cargo name', async () => {
      renderWithRouter(
        <AuthProvider>
          <OrdersProvider>
            <Orders />
          </OrdersProvider>
        </AuthProvider>
      );
      await waitFor(() => expect(ordersClient.getOrders).toBeCalledTimes(1));

      jest.clearAllMocks();

      fireEvent.click(screen.getAllByText('Cargo Name')[1]);

      await waitFor(() => {
        expect(ordersClient.getOrders).toBeCalledWith({
          filter: {},
          limit: DEFAULT_LIMIT,
          offset: 0,
          order: {
            cargoName: 'asc'
          }
        });
      });
    });

    it('should filter orders by city from', async () => {
      renderWithRouter(
        <AuthProvider>
          <OrdersProvider>
            <Orders />
          </OrdersProvider>
        </AuthProvider>
      );
      await waitFor(() => expect(ordersClient.getOrders).toBeCalledTimes(1));

      jest.clearAllMocks();

      fireEvent.click(screen.getAllByText('From')[1]);

      await waitFor(() => {
        expect(ordersClient.getOrders).toBeCalledWith({
          filter: {},
          limit: DEFAULT_LIMIT,
          offset: 0,
          order: {
            cityFrom: 'asc'
          }
        });
      });
    });

    it('should filter orders by city to', async () => {
      renderWithRouter(
        <AuthProvider>
          <OrdersProvider>
            <Orders />
          </OrdersProvider>
        </AuthProvider>
      );
      await waitFor(() => expect(ordersClient.getOrders).toBeCalledTimes(1));

      jest.clearAllMocks();

      fireEvent.click(screen.getAllByText('To')[1]);

      await waitFor(() => {
        expect(ordersClient.getOrders).toBeCalledWith({
          filter: {},
          limit: DEFAULT_LIMIT,
          offset: 0,
          order: {
            cityTo: 'asc'
          }
        });
      });
    });
  });

  describe('pagination', () => {
    it('should paginate orders', async () => {
      renderWithRouter(
        <AuthProvider>
          <OrdersProvider>
            <Orders />
          </OrdersProvider>
        </AuthProvider>
      );
      await waitFor(() => expect(ordersClient.getOrders).toBeCalledTimes(1));

      jest.clearAllMocks();

      fireEvent.click(screen.getByText('2'));

      await waitFor(() => {
        expect(ordersClient.getOrders).toBeCalledWith({
          filter: {},
          limit: DEFAULT_LIMIT,
          offset: 10,
          order: {
            id: 'desc'
          }
        });
      });
    });
  });
});
