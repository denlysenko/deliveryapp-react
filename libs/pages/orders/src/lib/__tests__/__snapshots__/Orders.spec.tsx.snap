// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders Page should render successfully 1`] = `
<body>
  <div>
    <div
      class="sc-AxhCb dyCIgT"
    >
      <div
        class="sc-AxirZ dkfEnX"
      >
        <div
          class="p-dialog-mask p-component-overlay p-dialog-center"
        />
      </div>
      <div
        class="card"
      >
        <div
          class="p-grid"
        >
          <div
            class="p-md-8 p-col-12"
          >
            <div
              class="sc-AxiKw kwGUfS"
            >
              <div
                class="order-filter p-grid"
              >
                <div
                  class="p-inputgroup p-col-12 p-md-6"
                >
                  <span
                    class="p-inputgroup-addon"
                  >
                    <i
                      class="fa fa-search"
                    />
                  </span>
                  <input
                    class="p-inputtext p-component"
                    data-testid="searchTerm"
                    placeholder="Search..."
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="p-col-12 p-md-6"
                >
                  <div
                    class="p-dropdown p-component p-inputwrapper-filled"
                  >
                    <div
                      class="p-hidden-accessible"
                    >
                      <input
                        aria-haspopup="listbox"
                        readonly=""
                        type="text"
                      />
                    </div>
                    <div
                      class="p-hidden-accessible p-dropdown-hidden-select"
                    >
                      <select
                        aria-hidden="true"
                        tabindex="-1"
                      >
                        <option
                          value=""
                        />
                        <option
                          value="id"
                        >
                          Order number
                        </option>
                      </select>
                    </div>
                    <label
                      class="p-dropdown-label p-inputtext"
                    >
                      Order number
                    </label>
                    <div
                      aria-haspopup="listbox"
                      class="p-dropdown-trigger"
                      role="button"
                    >
                      <span
                        class="p-dropdown-trigger-icon pi pi-chevron-down p-clickable"
                      />
                    </div>
                    <div
                      class="p-dropdown-panel p-hidden p-input-overlay"
                    >
                      <div
                        class="p-dropdown-items-wrapper"
                        style="max-height: 200px;"
                      >
                        <ul
                          class="p-dropdown-items p-dropdown-list p-component"
                          role="listbox"
                        >
                          <li
                            aria-label="Order number"
                            aria-selected="true"
                            class="p-dropdown-item p-highlight"
                            role="option"
                          >
                            Order number
                          </li>
                          <li
                            aria-label="Cargo Name"
                            aria-selected="false"
                            class="p-dropdown-item"
                            role="option"
                          >
                            Cargo Name
                          </li>
                          <li
                            aria-label="From"
                            aria-selected="false"
                            class="p-dropdown-item"
                            role="option"
                          >
                            From
                          </li>
                          <li
                            aria-label="To"
                            aria-selected="false"
                            class="p-dropdown-item"
                            role="option"
                          >
                            To
                          </li>
                        </ul>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="button p-md-4 p-col-12"
          >
            <a
              href="/orders/create"
            >
              <button
                class="p-button p-component raised-btn p-button-text-icon-left"
                type="button"
              >
                <span
                  class="fa fa-plus p-c p-button-icon-left"
                />
                <span
                  class="p-button-text p-c"
                >
                  Create order
                </span>
              </button>
            </a>
          </div>
        </div>
        <div
          class="p-datatable p-component"
        >
          <div
            class="p-datatable-wrapper"
          >
            <table
              role="grid"
            >
              <thead
                class="p-datatable-thead"
              >
                <tr>
                  <th
                    aria-sort="none"
                    class="p-sortable-column p-highlight id"
                    rowspan="2"
                  >
                    <span
                      class="p-column-title"
                    >
                      Num.
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-amount-down"
                    />
                  </th>
                  <th
                    aria-sort="none"
                    class="p-sortable-column cargoName"
                    rowspan="2"
                  >
                    <span
                      class="p-column-title"
                    >
                      Cargo Name
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                    />
                  </th>
                  <th
                    class=""
                    rowspan="2"
                  >
                    <span
                      class="p-column-title"
                    >
                      Weight/Volume
                    </span>
                  </th>
                  <th
                    class=""
                    colspan="2"
                  >
                    <span
                      class="p-column-title"
                    >
                      Route
                    </span>
                  </th>
                  <th
                    class=""
                    rowspan="2"
                  >
                    <span
                      class="p-column-title"
                    >
                      Delivery Costs
                    </span>
                  </th>
                  <th
                    class=""
                    rowspan="2"
                  >
                    <span
                      class="p-column-title"
                    >
                      Delivery Date
                    </span>
                  </th>
                  <th
                    class=""
                    rowspan="2"
                  >
                    <span
                      class="p-column-title"
                    >
                      Status
                    </span>
                  </th>
                  <th
                    class=""
                    rowspan="2"
                  >
                    <span
                      class="p-column-title"
                    >
                      Paid
                    </span>
                  </th>
                </tr>
                <tr>
                  <th
                    aria-sort="none"
                    class="p-sortable-column cityFrom"
                  >
                    <span
                      class="p-column-title"
                    >
                      From
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                    />
                  </th>
                  <th
                    aria-sort="none"
                    class="p-sortable-column cityTo"
                  >
                    <span
                      class="p-column-title"
                    >
                      To
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                    />
                  </th>
                </tr>
              </thead>
              <tbody
                class="p-datatable-tbody"
              >
                <tr
                  class="p-datatable-row"
                >
                  <td
                    class=""
                  >
                    <a
                      href="/orders/1"
                    >
                      1
                    </a>
                  </td>
                  <td
                    class=""
                  >
                    Computers
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      12 kg / - m3
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    Kyiv
                  </td>
                  <td
                    class=""
                  >
                    Dnipro
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      New
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      No
                    </span>
                  </td>
                </tr>
                <tr
                  class="p-datatable-row"
                >
                  <td
                    class=""
                  >
                    <a
                      href="/orders/1"
                    >
                      1
                    </a>
                  </td>
                  <td
                    class=""
                  >
                    Computers
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      12 kg / - m3
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    Kyiv
                  </td>
                  <td
                    class=""
                  >
                    Dnipro
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      New
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      No
                    </span>
                  </td>
                </tr>
                <tr
                  class="p-datatable-row"
                >
                  <td
                    class=""
                  >
                    <a
                      href="/orders/1"
                    >
                      1
                    </a>
                  </td>
                  <td
                    class=""
                  >
                    Computers
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      12 kg / - m3
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    Kyiv
                  </td>
                  <td
                    class=""
                  >
                    Dnipro
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      New
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      No
                    </span>
                  </td>
                </tr>
                <tr
                  class="p-datatable-row"
                >
                  <td
                    class=""
                  >
                    <a
                      href="/orders/1"
                    >
                      1
                    </a>
                  </td>
                  <td
                    class=""
                  >
                    Computers
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      12 kg / - m3
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    Kyiv
                  </td>
                  <td
                    class=""
                  >
                    Dnipro
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      New
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      No
                    </span>
                  </td>
                </tr>
                <tr
                  class="p-datatable-row"
                >
                  <td
                    class=""
                  >
                    <a
                      href="/orders/1"
                    >
                      1
                    </a>
                  </td>
                  <td
                    class=""
                  >
                    Computers
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      12 kg / - m3
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    Kyiv
                  </td>
                  <td
                    class=""
                  >
                    Dnipro
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      New
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      No
                    </span>
                  </td>
                </tr>
                <tr
                  class="p-datatable-row"
                >
                  <td
                    class=""
                  >
                    <a
                      href="/orders/1"
                    >
                      1
                    </a>
                  </td>
                  <td
                    class=""
                  >
                    Computers
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      12 kg / - m3
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    Kyiv
                  </td>
                  <td
                    class=""
                  >
                    Dnipro
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      New
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      No
                    </span>
                  </td>
                </tr>
                <tr
                  class="p-datatable-row"
                >
                  <td
                    class=""
                  >
                    <a
                      href="/orders/1"
                    >
                      1
                    </a>
                  </td>
                  <td
                    class=""
                  >
                    Computers
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      12 kg / - m3
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    Kyiv
                  </td>
                  <td
                    class=""
                  >
                    Dnipro
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      New
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      No
                    </span>
                  </td>
                </tr>
                <tr
                  class="p-datatable-row"
                >
                  <td
                    class=""
                  >
                    <a
                      href="/orders/1"
                    >
                      1
                    </a>
                  </td>
                  <td
                    class=""
                  >
                    Computers
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      12 kg / - m3
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    Kyiv
                  </td>
                  <td
                    class=""
                  >
                    Dnipro
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      New
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      No
                    </span>
                  </td>
                </tr>
                <tr
                  class="p-datatable-row"
                >
                  <td
                    class=""
                  >
                    <a
                      href="/orders/1"
                    >
                      1
                    </a>
                  </td>
                  <td
                    class=""
                  >
                    Computers
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      12 kg / - m3
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    Kyiv
                  </td>
                  <td
                    class=""
                  >
                    Dnipro
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      New
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      No
                    </span>
                  </td>
                </tr>
                <tr
                  class="p-datatable-row"
                >
                  <td
                    class=""
                  >
                    <a
                      href="/orders/1"
                    >
                      1
                    </a>
                  </td>
                  <td
                    class=""
                  >
                    Computers
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      12 kg / - m3
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    Kyiv
                  </td>
                  <td
                    class=""
                  >
                    Dnipro
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      New
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      No
                    </span>
                  </td>
                </tr>
                <tr
                  class="p-datatable-row"
                >
                  <td
                    class=""
                  >
                    <a
                      href="/orders/1"
                    >
                      1
                    </a>
                  </td>
                  <td
                    class=""
                  >
                    Computers
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      12 kg / - m3
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    Kyiv
                  </td>
                  <td
                    class=""
                  >
                    Dnipro
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      New
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      No
                    </span>
                  </td>
                </tr>
                <tr
                  class="p-datatable-row"
                >
                  <td
                    class=""
                  >
                    <a
                      href="/orders/1"
                    >
                      1
                    </a>
                  </td>
                  <td
                    class=""
                  >
                    Computers
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      12 kg / - m3
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    Kyiv
                  </td>
                  <td
                    class=""
                  >
                    Dnipro
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span />
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      New
                    </span>
                  </td>
                  <td
                    class=""
                  >
                    <span>
                      No
                    </span>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="p-paginator p-component p-unselectable-text"
        >
          <button
            class="p-paginator-first p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-paginator-icon pi pi-step-backward"
            />
          </button>
          <button
            class="p-paginator-prev p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-paginator-icon pi pi-caret-left"
            />
          </button>
          <span
            class="p-paginator-pages"
          >
            <button
              class="p-paginator-page p-paginator-element p-link p-highlight"
              type="button"
            >
              1
            </button>
            <button
              class="p-paginator-page p-paginator-element p-link"
              type="button"
            >
              2
            </button>
          </span>
          <button
            class="p-paginator-next p-paginator-element p-link"
            type="button"
          >
            <span
              class="p-paginator-icon pi pi-caret-right"
            />
          </button>
          <button
            class="p-paginator-last p-paginator-element p-link"
            type="button"
          >
            <span
              class="p-paginator-icon pi pi-step-forward"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
